modules/ROOT/pages/6.1-release-notes.adoc

```diff
@@ -0,0 +1,1121 @@
+
+With this update, {productname} now supports applying, altering and removing `align` styles to text set in `+<pre>+` tags.
+
+
+=== Nuget packages for .NET and .NET Core are now configured to copy TinyMCE into `/wwwroot/lib/` when TinyMCE is installed into a project
+
+Previously, the {productname} Nuget package did not automatically copy {productname} in to `/wwwroot/lib/` when installing {productname} into a .NET project.
+
+This resulted in {productname} not working as expected (or at all).
+
+This error has been corrected with this update. Installing the {productname} Nuget package into a .NET project automatically places {productname} in `/wwwroot/lib`, as expected.
+
+
+=== UI components, such as dialogs, would in some cases cause the esc keyup event to incorrectly trigger inside the editor
+
+In {productname} the *esc* (Escape) key can be used to exit or ‘escape’ from an active user-interface component. If a menu is open, pressing the esc key closes the menu. If a dialog is present, pressing the esc closes the dialog.
+
+{productname} responds to the `KeyDown` event and exits the active element when the *esc* key is pressed. Previously, however, the consequent `KeyUp` event (generated when the user stops pressing the *esc* key) was propogated to the {productname} editor. This should not have happened.
+
+The `KeyDown` event does not propagate outside the component listening for it and neither should the `KeyUp` event.
+
+As of this update, this is the behavior: pressing the *esc* key closes or exits an active user-interface component and neither the `KeyDown` or `KeyUp event are propogated outside the component listening for these events.
+
+
+=== Selection direction was not stored/restored when getting/setting selection bookmarks
+
+_Selection direction_ is a real-time state that ‘knows’ where the current text selection began and, consequently, ‘knows’ which direction of pointer movement or which arrow key selects more text or de-selects already selected text.
+
+For example, consider a document containing words written in a left-to-right script and an insertion point set at the end of a word. A user then holds down the Shift key and presses the left-arrow key. The selection direction for this moment and this action is set by said action. While the Shift key is held down, pressing the left-arrow key adds letters to the selection and pressing the right-arrow key removes letters from the selection.
+
+When a {productname} document had *View > Show invisible characters* checked and text was selected in this manner, and the user paused for a second or two, pausing the selection extentsion, the `moveToBookmark` function copied the current selection into a bookmark and then restored the selection in the document.
+
+In doing this, however, the selection direction was lost, because `movetoBookmark` did not store the selection direction state.
```

Comment:
```suggestion
In doing this, however, the selection direction was lost, because `moveToBookmark` did not store the selection direction state.
```

Reply:
Also this isn't technically correct... it's `getBookmark` that didn't store the selection direction, meaning `moveToBookmark` couldn't restore it.